src/main/scripts/org/reaktivity/specification/http2/rfc7540/connection.management/client.sent.read.abort.on.closed.request/server.rpt

```diff
@@ -14,11 +14,13 @@
 # under the License.
 #
 
-property newServerAcceptRef ${nukleus:newRouteRef()} # external
+property newServerAcceptRef ${nuklei:newReferenceId()} # external
```

Comment:
Are you using the property anymore ?

Reply:
Not sure what is happening, but it seems that the function called will define the type of the variable. One returns a long, another byte array, if I remember correctly. With incorrect type, the route is not working properly, because newServerAcceptRef is a byte array in controller.rpt.

Reply:
newServerAcceptReference isn't used in this script so it shouldn't be needed I would think.  This is probably just a mistake in the matching script in http: https://github.com/reaktivity/nukleus-http.spec/blob/develop/src/main/scripts/org/reaktivity/specification/http/rfc7230/connection.management/concurrent.upgrade.requests.and.responses.with.data/server.rpt

Reply:
It is used in the client.rpt script. The combination:
```
property newServerAcceptRef ${nuklei:newReferenceId()} # external
property newClientConnectRef ${newServerAcceptRef}
```
is required because one variable is used in the spec, the other is used in the java module. All scripts in spec have this combination. And without it, the client tests in java are not working at all.

Reply:
newServerAcceptRef is not used in the script so I would expect it is not needed here.  I would also expect that is a typo in http: https://github.com/reaktivity/nukleus-http.spec/blob/develop/src/main/scripts/org/reaktivity/specification/http/rfc7230/message.format/incomplete.request.with.unimplemented.method/server.rpt.

@cmebarrow is this correct?

Reply:
I have tried to delete that line and the test in java is failing. There are 4 variables defined in controller.rpt (two in client mode, two in server mode). They are used differently in spec and in java.

Reply:
Which test?  perhaps that will provide insight into why it is needed and answer Jitu's original question.

Reply:
I will try to explain how these tests are working. Each test case will be represented by 4 tests in nuklei that have client/server mode. Two will be in the specs (rfc and nukleus), two in the java module (client mode and server mode).

-in java for client mode:
	-we run client/controller.rpt, nukleus/client.rpt and rfc/server.rpt
	-controller.rpt defines `newClientConnectRef` and reads `newClientAcceptRef` from the answer
	-nukleus/client.rpt must use `newClientAcceptRef`
	-rfc/server.rpt must use `newClientConnectRef` (for example, specification\http2\rfc7540\connection.abort\client.sent.read.abort.on.open.request.response.buffered\server.rpt will not work in this configuration, it uses only `newServerAcceptRef`, valid for spec test)

-in java for server mode:
	-we run controller.rpt, rfc/client.rpt and nukleus/server.rpt
	-controller.rpt defines `newServerConnectRef` and reads `newServerAcceptRef` from the answer
	-rfc/client.rpt must use `newServerAcceptRef`
	-nukleus/server.rpt must use `newServerConnectRef`

-in spec for rfc:
	-rfc/server.rpt runs together with rfc/client.rpt.
	-both scripts need to use the same route ref. Because of this, the variables from above, `newClientConnectRef` and `newServerAcceptRef` must have the same value in this test

-in spec for nukleus:
	-nukleus/server.rpt runs together with nukleus/client.rpt.
	-similarly, both scripts need to use the same route ref. Because of this, the variables from above, `newServerConnectRef` and `newClientAcceptRef` must have the same value in this test

in order to run without the controller.rpt in spec, those two variables must be defined also in client/server scripts. Because in all scripts (even controller) the connect ref is initialized with some random value, it does not matter if it is initialized twice. As k3po is quite limited, this artifice of declaring and overwriting the value was used.

Reply:
The newServerAcceptRef is generated by top level routes and passed into other routes.  For example, here is binding echo on WS.

```
route "http" 0 "ws" 0 echo
route "tcp" 0 "ws" GENERATED_SOURCE_FROM_LAST_ROUTE_COMMAND
route "any" port "http" GENERATED_SOURCE_2_FROM_LAST_ROUTE_COMMAND
```

Here, the GENERATED_SOURCE_FROM_LAST_ROUTE_COMMAND is the newServerAcceptRef.  It is in the server.rpt because in the tests there is no real server (existing route) that is created.